src/commands/registries/logOutOfDockerCli.ts

```diff
@@ -6,26 +6,16 @@
 import { Terminal } from 'vscode';
 import { IActionContext } from 'vscode-azureextensionui';
 import { ext } from '../../extensionVariables';
-import { DockerHubAccountTreeItem } from '../../tree/registries/dockerHub/DockerHubAccountTreeItem';
-import { DockerHubNamespaceTreeItem } from '../../tree/registries/dockerHub/DockerHubNamespaceTreeItem';
+import { registryExpectedContextValues } from '../../tree/registries/registryContextValues';
 import { RegistryTreeItemBase } from '../../tree/registries/RegistryTreeItemBase';
 
-export async function logOutOfDockerCli(context: IActionContext, node?: RegistryTreeItemBase | DockerHubAccountTreeItem): Promise<void> {
+export async function logOutOfDockerCli(context: IActionContext, node?: RegistryTreeItemBase): Promise<void> {
     if (!node) {
-        node = await ext.registriesTree.showTreeItemPicker<RegistryTreeItemBase | DockerHubAccountTreeItem>([DockerHubAccountTreeItem.contextValue, /^(azure|private)Registry$/i], context);
-    }
-
-    let command = 'docker logout';
-
-    if (node instanceof DockerHubNamespaceTreeItem) {
-        node = node.parent;
-    }
-
-    if (!(node instanceof DockerHubAccountTreeItem)) {
-        command = `${command} ${node.baseUrl}`;
+        node = await ext.registriesTree.showTreeItemPicker<RegistryTreeItemBase>(registryExpectedContextValues.all.registry, context);
     }
 
+    let creds = await node.getDockerCliCredentials();
     const terminal: Terminal = ext.terminalProvider.createTerminal('docker logout');
-    terminal.sendText(command);
+    terminal.sendText(`docker logout ${creds.registryPath}`);
```

Comment:
Does it make sense to replace all of the terminal calls with `dockerode`?

Reply:
I've considered it, but that's a pretty big refactor and I've probably already gone too far down the refactoring rabbit-hole lately